### PR TITLE
Pipelines build-info-go release

### PR DESCRIPTION
**Pull request by gailazar300**
- [x] All [tests](https://github.com/jfrog/build-info-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----
Build and release build-info-go cli binaries.

### CODE REVIEW

File: pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+resources:
+- name: buildinfoGoGit
+  type: GitRepo
+  configuration:
+    path: jfrog/build-info-go
+    branches:
+      include: main
+    gitProvider: il_automation
```
Comment (yahavi) [inline pipelines.yml:8]:
```suggestion
  - name: buildinfoGoGit
    type: GitRepo
    configuration:
      path: jfrog/build-info-go
      branches:
        include: main
      gitProvider: il_automation
```

File: pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        readOnly:
+          NEXT_VERSION: 0.0.0
+
+    steps:
+      - name: Release
+        type: Bash
+        configuration:
+          inputResources:
+            - name: buildinfoGoGit
+              trigger : false
```
Comment (yahavi) [inline pipelines.yml:29]:
```suggestion
              trigger: false
```

File: release/build.sh [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+version="$1"
+pkgPath="ecosys-bi-cli/v1"
+
+# Verify version provided in pipelines UI matches version in build-info-go source code.
+verifyVersionMatching
+
+# Build and upload for every architecture.
+# Keep 'linux-386' first to prevent unnecessary uploads in case the built version doesn't match the provided one.
```
Comment (yahavi) [inline release/build.sh:69]: What's special in linux-386? I couldn't understand the reason

**Review by yahavi — CHANGES_REQUESTED** [review]

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
Let's also modify the binaries release logic, to avoid deployment failures, as a result of the new binaries overriding the old ones.

**Pull request closed without merging**